requests/request.go

```diff
@@ -91,6 +91,11 @@ func (request *Request) SetTimeout(timeout int16) {
 func processResponse(response *http.Response) (map[string]interface{}, error) {
 	defer response.Body.Close()
 	body, err := ioutil.ReadAll(response.Body)
+
+	if len(body) == 0 || len(body) == 2{
+        resp :=  make(map[string]interface{})
+        return resp, nil
+    }
```

Comment:
Why this change ?

Reply:
because some of api's response is empty array and with this condition we can handle empty array

Reply:
@ankitdas13 With this **len(body) == 2** we will be sending empty empty array even when the length is 2, is this expected ?

Reply:
@SushanthGunjal 
1 . In this [api](https://razorpay.com/docs/smart-collect/api/delete-tpv/) if you check the response it returns `null`.  for handling this i have added `len(body)==0`

2. In this [api](https://api.razorpay.com/v1/items/item_7Oy8OMV6BdEAac) we are getting empty array where `len(body)==2` count that bracket .